src/coroutine/memoset/mod.rs

```diff
@@ -567,22 +567,31 @@ impl<F: LurkField, Q: Query<F>> Scope<Q, LogMemo<F>, F> {
     // }
 
     fn compute_provenances(&self, store: &Store<F>) -> HashMap<ZPtr<Tag, F>, ZPtr<Tag, F>> {
-        let base_cases = self
+        let mut provenances = HashMap::default();
+        let mut ready = HashSet::new();
+
+        let mut missing_dependency_counts: HashMap<&Ptr, usize> = self
             .queries
             .keys()
-            .filter(|key| {
-                if let Some(deps) = self.dependencies.get(key) {
-                    deps.is_empty()
-                } else {
-                    true
+            .map(|key| {
+                let dep_count = self.dependencies.get(key).map_or(0, |x| x.len());
+
+                if dep_count == 0 {
+                    // Queries are ready if they have no missing dependencies.
+                    // Initially, this will be the base cases -- which have no dependencies.
+                    ready.insert(key);
                 }
+                (key, dep_count)
```

Comment:
When a query is ready here, we still include `(key, 0)` in `missing_dependency_counts`...

Reply:
Yep, a ready query has no missing dependencies. This also means that no other queries (its formerly missing dependencies) will trigger modification of that stat in the future.